### ISSUE TITLE
Trailing space on open/close of frontmatter causes MD003 error

### ISSUE THREAD

**Issue by nschonni** [DavidAnson/markdownlint#208]
EX:
```md
--- 
title: foo
---

# Bar
```
Trailing space after the `---` appears to cause Markdownlint to think that the Header style is setext.

**Comment by DavidAnson** [comment]
This behavior is deliberately strict, but perhaps a tad overly so.

The default regular expression for front matter prohibits spaces as you say, but can be customized by the user if desired: https://github.com/DavidAnson/markdownlint#optionsfrontmatter

I think the most authoritative source for behavior is the Jekyll blog engine: https://jekyllrb.com/docs/front-matter/

This is the regular expression that Jekyll uses: https://github.com/jekyll/jekyll/blob/13d31c4c8be13ab1536a0139a48952b142e81682/lib/jekyll/document.rb#L13

It does not support the `+++/...` syntax (used by other engines), but does allow spaces. I will look into relaxing the default behavior to allow spaces in each of these scenarios.

Thank you!

**Comment by nschonni** [comment]
Sounds good! I usually just trim it, but it can be confusing to get MD003 errors when the root issue is frontmatter space 😄
